internal/controllers/machine/machine_controller_test.go

```diff
@@ -2530,6 +2541,7 @@ func TestIsDeleteNodeAllowed(t *testing.T) {
 		name          string
 		cluster       *clusterv1.Cluster
 		machine       *clusterv1.Machine
+		infraMachine  *InfrastructureMachine
```

Comment:
```suggestion
		infraMachine  *unstructured.Unstructured
```

Let's use unstructured instead.